metals/src/main/scala/scala/meta/internal/metals/Compilers.scala

```diff
@@ -113,14 +122,80 @@ class Compilers(
       }
     }
 
+  def foldingRange(
+      params: FoldingRangeRequestParams,
+      token: CancelToken
+  ): Future[ju.List[FoldingRange]] = {
+    val path = params.getTextDocument.getUri.toAbsolutePath
+    val pc = loadCompiler(path, None).getOrElse(ramboCompiler)
+    val input = path.toInputFromBuffers(buffers)
+    pc.foldingRange(
+        CompilerVirtualFileParams(path.toNIO.toUri, input.value)
+      )
+      .asScala
+  }
+
+  def onTypeFormatting(
+      params: DocumentOnTypeFormattingParams
+  ): Future[ju.List[TextEdit]] = {
+    val path = params.getTextDocument.getUri.toAbsolutePath
+    val pc = loadCompiler(path, None).getOrElse(ramboCompiler)
+    val input = path.toInputFromBuffers(buffers)
+    pc.onTypeFormatting(params, input.value).asScala
+  }
+
+  def rangeFormatting(
+      params: DocumentRangeFormattingParams
+  ): Future[ju.List[TextEdit]] = {
+    val path = params.getTextDocument.getUri.toAbsolutePath
+    val pc = loadCompiler(path, None).getOrElse(ramboCompiler)
+    val input = path.toInputFromBuffers(buffers)
+    pc.rangeFormatting(params, input.value).asScala
+  }
+
+  def documentSymbol(
+      params: DocumentSymbolParams
+  ): Future[ju.List[DocumentSymbol]] = {
+    val path = params.getTextDocument.getUri.toAbsolutePath
+    val pc = loadCompiler(path, None).getOrElse(ramboCompiler)
+    val input = path.toInputFromBuffers(buffers)
+    pc.documentSymbols(
+        CompilerVirtualFileParams(path.toNIO.toUri, input.value)
+      )
+      .asScala
+  }
+
+  def didClose(path: AbsolutePath): Unit = {
+    val pc = loadCompiler(path, None).getOrElse(ramboCompiler)
+    pc.didClose(path.toNIO.toUri())
+  }
+
+  def didChange(path: AbsolutePath): Future[Unit] = {
+    val pc = loadCompiler(path, None).getOrElse(ramboCompiler)
+    val input = path.toInputFromBuffers(buffers)
+    for {
+      ds <- pc
+        .didChange(CompilerVirtualFileParams(path.toNIO.toUri(), input.value))
+        .asScala
+    } yield {
+      ds.asScala.headOption match {
+        case None =>
+          diagnostics.onNoSyntaxError(path)
+        case Some(diagnostic) =>
+          diagnostics.onSyntaxError(path, diagnostic)
+      }
+    }
+  }
+
   def didCompile(report: CompileReport): Unit = {
     if (report.getErrors > 0) {
       cache.get(report.getTarget).foreach(_.restart())
     } else {
       // Restart PC for all build targets that depend on this target since the classfiles
       // may have changed.
       for {
-        target <- buildTargets.inverseDependencies(report.getTarget)
+        target <- buildTargets.inverseDependencies(report.getTarget) + report
```

Comment:
InverseDependencies doesn't contain the current target,

Reply:
Hmm, I wonder if this change fixes weird issues I've hit on where "connect to build server" have fixed it 🤔 

nit: please try to manually break on infix operators
```suggestion
        target <- buildTargets.inverseDependencies(report.getTarget) +
          report.getTarget()
```

Reply:
Fixed!